Sources/Purchasing/ProductsManager.swift

```diff
@@ -74,9 +74,7 @@ class ProductsManager: NSObject {
     func clearCachedProducts() {
         if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *),
            self.systemInfo.storeKit2Setting == .enabledForCompatibleDevices {
-            Task<Void, Never> {
-                await productsFetcherSK2.clearCache()
-            }
+            // We have to re-fetch the products
```

Comment:
I wanted to use `#unavailable` here, but I decided to leave this logic because we're going to re-fetch the products in the next step

Reply:
is the implementation of this coming in a separate PR?

Reply:
yep, the logic was to be included in step 3 but finally we decided to modify the strategy. It's related to `[this thread](https://github.com/RevenueCat/purchases-ios/pull/1583#discussion_r878370125)`

Reply:
could we point this PR to an integration brach? So that we have `main` always in a consistent state - it might otherwise be daunting if we have a bug report and when investigating we see that `clearCachedProducts` has been called, we might reasonably assume that cache has been cleared unless we see that the method is a no-op for sk2